src/components/EventList.tsx

```diff
@@ -5,36 +5,53 @@ import { Context } from "../store/app-context";
 import { timeTransformer } from "../utils/timeTransformer";
 import useFireStore from "../hooks/useFireStore";
 import { Event } from "./Pie";
+import { toast } from "react-toastify";
+import WarningModal from "./WarningModal";
 
 const EventList: React.FC = () => {
+  const [isShowModal, setIsShowModal] = useState(false);
   const [deleting, setDeleting] = useState(false);
   const [selectedId, setSelectedId] = useState("");
   const firestore = useFireStore();
   const { date, data, setData } = useContext(Context);
 
   const onClickHandler = (e: React.MouseEvent) => {
-    if (!deleting) {
+    if (e.currentTarget.id === selectedId && deleting) {
+      setDeleting(false);
+    } else {
       setDeleting(true);
       setSelectedId(e.currentTarget.id);
-    } else {
-      setDeleting(false);
-      setSelectedId("");
     }
   };
 
   const deleteEventHandler = () => {
-    const newData = data.filter((elem) => elem.id !== selectedId);
-
-    setData(newData);
-    sendData(newData);
+    setIsShowModal(true);
```

Comment:
could make a function like toggleModalShow()=>setIsShowModal(!isShowModal) and pass it to modal onDecline, onConfirmHandler, Delete button